### PR TITLE
MGDOBR-1266 Migrate GitHub Actions to use AWS runner

### PR DESCRIPTION
**Pull request by sutaakar**
Signed-off-by: Karel Suta <ksuta@redhat.com>

[MGDOBR-1266](https://issues.redhat.com/browse/MGDOBR-1266)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: .github/workflows/Manager-CI.yaml
Comment (danielezonca) [inline .github/workflows/Manager-CI.yaml:2]: Side note for the future: I think the `Manager-CI` job should be able to run independently without the need of the full data plane. We might review this part 🤔

**Review by danielezonca — COMMENTED** [review]
Thanks for the PR, well done :)
I know it is still draft but I added a couple of comments.
Btw don't wait for my approval :)

File: .github/actions/kind/action.yml
Comment (sutaakar) [reply .github/actions/kind/action.yml:24]: `kind` is currently preinstalled in the runner image

File: .github/actions/kind/action.yml
Comment (sutaakar) [reply .github/actions/kind/action.yml:14]: I hit an issue using this action when running on runner using AWS linux:
```
Run container-tools/kind-action@v2
  with:
    config: .github/kind-config.yaml
    registry: true
    node_image: kindest/node:v1.22.2
  env:
    NAMESPACE: mynamespace
    KIND_MAIN_NODE_HOSTNAME: kind-control-plane
    JAVA_HOME: /actions-runner/_work/_tool/Java_Temurin-Hotspot_jdk/11.0.17-8/x64
    JAVA_HOME_11_X64: /actions-runner/_work/_tool/Java_Temurin-Hotspot_jdk/11.0.17-8/x64
/actions-runner/_work/_actions/container-tools/kind-action/v2/main.sh: line 66: args[@]: unbound variable
```
Haven't tried to fix or report this. It may be worth reporting and in case it is fixed to migrate back to this action. As a workaround we can use current approach of installing KinD by ourselves.

File: .github/actions/kind/action.yml
Comment (sutaakar) [reply .github/actions/kind/action.yml:36]: After my changes the line below was failing on non existing pod, I guess it was caused by KinD not being fully ready yet (maybe because now we don't wait for KinD to be running?). This line makes sure that Ingress Deployment is ready

File: .github/workflows/Manager-CI.yaml
Comment (sutaakar) [reply .github/workflows/Manager-CI.yaml:2]: Depends on what should we test there.
For testing just control plane in isolation we can leverage Quarkus tests, done as part of unit testing.

File: .github/workflows/Operator-CI.yaml
Comment (danielezonca) [inline .github/workflows/Operator-CI.yaml:31]: Same comment here

File: .github/workflows/Upgrade-CI.yml
Comment (danielezonca) [inline .github/workflows/Upgrade-CI.yml:6]: To be changed before merging

File: .github/workflows/Upgrade-CI.yml
Comment (danielezonca) [inline .github/workflows/Upgrade-CI.yml:331]: Same here

File: .github/workflows/Upgrade-CI.yml
Comment (danielezonca) [inline .github/workflows/Upgrade-CI.yml:47]: Same here

File: .github/workflows/Upgrade-CI.yml
Comment (danielezonca) [inline .github/workflows/Upgrade-CI.yml:45]: Same here

**Review by danielezonca — COMMENTED** [review]
Some minor comments

File: .github/workflows/Manager-CI.yaml
Comment (sutaakar) [reply .github/workflows/Manager-CI.yaml]: Done

File: .github/workflows/Manager-CI.yaml
Comment (sutaakar) [reply .github/workflows/Manager-CI.yaml:42]: The check was moved to `start-runner`, so in case the label is not provided then the EC2 instance isn't provisioned.

File: .github/workflows/Operator-CI.yaml
Comment (sutaakar) [reply .github/workflows/Operator-CI.yaml:31]: Same as above

File: .github/workflows/Upgrade-CI.yml
Comment (sutaakar) [reply .github/workflows/Upgrade-CI.yml:47]: Same as above

File: .github/workflows/Upgrade-CI.yml
Comment (sutaakar) [reply .github/workflows/Upgrade-CI.yml]: Fixed

File: .github/actions/start-aws-runner/action.yml
Comment (sutaakar) [reply .github/actions/start-aws-runner/action.yml:7]: right

Comment (sutaakar) [reply .github/actions/start-aws-runner/action.yml:7]:
```
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests      Limits
  --------           --------      ------
  cpu                2020m (50%)   2700m (67%)
  memory             3176Mi (20%)  2814Mi (17%)
  ```
  unfortunately I cannot get exact actual usage in node
  this is for the old installation, without Camel K changes

**Review by r00ta — APPROVED** [review]

**Review by kostola — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

**Pull request merged**
